spec/controllers/campaigns_controller_spec.rb

```diff
@@ -123,4 +123,128 @@
       end
     end
   end
+
+  describe 'PUT #publish' do
+    before do
+      put :publish, params: { id: campaign.id }
+    end
+
+    context 'when state is new' do
+      it { expect(campaign).to have_state(:new) }
+
+      it { expect(campaign).not_to have_state(:on_hold) }
```

Comment:
those 2 specs are redundant, it is impossible to be in a few states in the same time.